### PR TITLE
[media] Waiting for setting CDM before initializing StarboardRenderer

### PR DESCRIPTION
**Pull request by borongc**
1. Follow Chromium's `RendererImpl` to wait for setting CDM in `StarboardRenderer::Initialize()`, so Cobalt can ensure CDM is set before `CreatePlayerBridge()`.
2. Add `state_` to ensure `CreatePlayerBridge()` will only be called once. 

b/375273774
b/378958007

### CODE REVIEW

File: media/starboard/starboard_renderer.cc [hunk 1]
```diff
@@ -183,16 +183,29 @@ void StarboardRenderer::SetCdm(CdmContext* cdm_context,
     return;
   }
 
-  drm_system_ = cdm_context->GetSbDrmSystem();
+  cdm_context_ = cdm_context;
+  drm_system_ = cdm_context_->GetSbDrmSystem();
   std::move(cdm_attached_cb).Run(true);
   LOG(INFO) << "CDM set successfully.";
+
+  if (state_ != STATE_INIT_PENDING_CDM) {
```
Comment (jasonzhangxx) [inline media/starboard/starboard_renderer.cc:191]: Could `state_` be `STATE_UNINITIALIZED` here?

File: media/starboard/starboard_renderer.cc [hunk 2]
```diff
@@ -183,16 +183,29 @@ void StarboardRenderer::SetCdm(CdmContext* cdm_context,
...
+
+  DCHECK(init_cb_);
+  state_ = STATE_INITIALIZING;
+  CreatePlayerBridge();
 }
 
 void StarboardRenderer::Flush(base::OnceClosure flush_cb) {
   DCHECK(task_runner_->RunsTasksInCurrentSequence());
   DCHECK(!pending_flush_cb_);
   DCHECK(flush_cb);
```
Comment (jasonzhangxx) [inline media/starboard/starboard_renderer.cc:204]: Are we confident to not reset `state_` in `Flush()`?

Comment (borongc) [reply media/starboard/starboard_renderer.cc:204]: Added `STATE_FLUSHED` after initialization or after flush completed.

File: media/starboard/starboard_renderer.cc (hunk 1 above)
Comment (borongc) [reply media/starboard/starboard_renderer.cc:191]: If it is `STATE_UNINITIALIZED`, then we let `Initialize()` to create player bridge. On Chromium, `SetCdm` is called before `Initialize`.

https://source.chromium.org/chromium/chromium/src/+/main:media/base/pipeline_impl.cc;l=1186;drc=5d0194cbc9cddc2ad096da32f16d7d1eac5d4a34;bpv=0;bpt=1

**Review by jasonzhangxx — APPROVED** [review]

**Pull request merged**